src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Model/Data/Query/ObservationQueryContextTests.cs

```diff
@@ -0,0 +1,116 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data.Query;
+using Xunit;
+using static GovUk.Education.ExploreEducationStatistics.Common.Model.TimeIdentifier;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.CollectionUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Model.Data.Query
+{
+    public class ObservationQueryContextTests
+    {
+        [Fact]
+        public void Clone_CreatesDeepCopy()
+        {
+            var subjectId = Guid.NewGuid();
+            var indicatorId = Guid.NewGuid();
+            var filterItemId = Guid.NewGuid();
+            var locationId = Guid.NewGuid();
+
+            var original = new ObservationQueryContext
+            {
+                SubjectId = subjectId,
+                BoundaryLevel = 1,
+                IncludeGeoJson = false,
+                Filters = ListOf(filterItemId),
+                Indicators = ListOf(indicatorId),
+                LocationIds = ListOf(locationId),
+                Locations = new LocationQuery
+                {
+                    Country = ListOf("original"),
+                    EnglishDevolvedArea = ListOf("original"),
+                    Institution = ListOf("original"),
+                    LocalAuthority = ListOf("original"),
+                    LocalAuthorityDistrict = ListOf("original"),
+                    LocalEnterprisePartnership = ListOf("original"),
+                    MultiAcademyTrust = ListOf("original"),
+                    MayoralCombinedAuthority = ListOf("original"),
+                    OpportunityArea = ListOf("original"),
+                    ParliamentaryConstituency = ListOf("original"),
+                    Provider = ListOf("original"),
+                    PlanningArea = ListOf("original"),
+                    Region = ListOf("original"),
+                    RscRegion = ListOf("original"),
+                    School = ListOf("original"),
+                    Sponsor = ListOf("original"),
+                    Ward = ListOf("original")
+                },
+                TimePeriod = new TimePeriodQuery
+                {
+                    StartCode = CalendarYear,
+                    StartYear = 2022,
+                    EndCode = CalendarYear,
+                    EndYear = 2022
+                }
+            };
+
+            var clone = original.Clone();
```

Comment:
Was assuming in this test that the first things we'd assert is that all of the fields in the clone were the same as the original, prior to the updates that we make to the clone.

I.e. pretty much a copy of all the assertions made on line 88 onwards, but applied to the clone directly after it's been created.

Reply:
Good point, I totally missed that we might not be cloning the original correctly. I've made an update in https://github.com/dfe-analytical-services/explore-education-statistics/pull/3189 to test this using `AssertDeepEquals` before any updates are made to the clone.